ecosystem/web-wallet/src/utils/account.ts

```diff
@@ -26,3 +26,29 @@ export function createNewAccount (): Result<AptosAccount, Error> {
   // todo: make request to create account on chain
   return ok(account)
 }
+
+export function getLocalStorageState (): LocalStorageState | null {
+  if (typeof window === 'undefined') {
+    return null
+  }
+  try {
+    // Get from local storage by key
+    const item = window.localStorage.getItem(walletStateLocalStorageKey)
+    return item ? JSON.parse(item) : null
+  } catch (_err) {
+    return null
+  }
+}
+
+export function getAptosAccountState (): AptosAccountState {
+  try {
+    const localStorage = getLocalStorageState()
+    if (localStorage) {
+      return AptosAccount.fromAptosAccountObject(localStorage as AptosAccountObject)
```

Comment:
> localStorage as AptosAccountObject

I think the localStorage object is being written wrong. It is actually a AptosAccountObject in local storage rather than a wrapper type

Reply:
You may want to add `(if typeof <expected property> === <expected type>)` checks here. Since TS uses structural typing, that should allow you to skip the type cast.

Reply:
I just fixes localStorage returning the wrong type, will need to discuss this later probably.